hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/S3Clients.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.hmpps.sqs
+
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.runBlocking
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest
+import software.amazon.awssdk.services.s3.model.ListBucketsRequest
+import uk.gov.justice.hmpps.sqs.telemetry.TraceInjectingExecutionInterceptor
+import java.net.URI
+
+internal fun createLocalstackS3AsyncClient(
+  localstackUrl: String,
+  region: String,
+  propagateTracing: Boolean,
+  bucketName: String,
+): S3AsyncClient = S3AsyncClient.builder()
+  .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("any", "any")))
+  .endpointOverride(URI.create(localstackUrl))
+  .forcePathStyle(true)
+  .region(Region.of(region))
+  .apply {
+    if (propagateTracing) {
+      overrideConfiguration { it.addExecutionInterceptor(TraceInjectingExecutionInterceptor()) }
+    }
+  }
+  .build()
+  .also {
+    runBlocking {
+      if (it.listBuckets(ListBucketsRequest.builder().build()).await().buckets().none { it.name() == bucketName }) {
```

Comment:
Not sure I understand the defensive clause here... If you've just built the S3 client why would it already know about the bucket?